From: Alexander Mikhaylenko <alexander.mikhaylenko@puri.sm>
Date: Thu, 24 Sep 2020 12:06:38 +0500
Subject: librem5: prefs-general-page: Redesign web application preferences

Split the grid into 3 separate rows for icon, title and homepage, shrink
the icon to 32x32.

Upstreamed in https://gitlab.gnome.org/GNOME/epiphany/-/merge_requests/834
---
 src/preferences/prefs-general-page.c    |   2 +-
 src/resources/gtk/prefs-general-page.ui | 116 ++++++++++++++------------------
 2 files changed, 50 insertions(+), 68 deletions(-)

diff --git a/src/preferences/prefs-general-page.c b/src/preferences/prefs-general-page.c
index 9d6460b..18fc680 100644
--- a/src/preferences/prefs-general-page.c
+++ b/src/preferences/prefs-general-page.c
@@ -865,7 +865,7 @@ prefs_general_page_update_webapp_icon (PrefsGeneralPage *general_page,
 
   icon_width = gdk_pixbuf_get_width (icon);
   icon_height = gdk_pixbuf_get_height (icon);
-  scale = MIN ((double)64 / icon_width, (double)64 / icon_height);
+  scale = MIN ((double)32 / icon_width, (double)32 / icon_height);
   scaled_icon = gdk_pixbuf_scale_simple (icon, icon_width * scale, icon_height * scale, GDK_INTERP_NEAREST);
   g_object_unref (icon);
   gtk_image_set_from_pixbuf (GTK_IMAGE (general_page->webapp_icon), scaled_icon);
diff --git a/src/resources/gtk/prefs-general-page.ui b/src/resources/gtk/prefs-general-page.ui
index bd49a3d..e3bd56f 100644
--- a/src/resources/gtk/prefs-general-page.ui
+++ b/src/resources/gtk/prefs-general-page.ui
@@ -10,79 +10,54 @@
         <property name="title" translatable="yes">Web Application</property>
         <property name="visible">True</property>
         <child>
-          <object class="HdyPreferencesRow">
+          <object class="HdyActionRow">
             <property name="activatable">False</property>
+            <property name="title" translatable="yes">_Icon</property>
+            <property name="use_underline">True</property>
             <property name="visible">True</property>
             <child>
-              <object class="GtkGrid">
-                <property name="column-spacing">12</property>
-                <property name="margin-bottom">8</property>
-                <property name="margin-end">12</property>
-                <property name="margin-start">12</property>
-                <property name="margin-top">8</property>
-                <property name="row-spacing">6</property>
-                <property name="visible">True</property>
-                <child>
-                  <object class="GtkButton" id="webapp_icon_button">
-                    <property name="visible">true</property>
-                    <property name="halign">center</property>
-                    <signal name="clicked" handler="on_webapp_icon_button_clicked"/>
-                    <child>
-                      <object class="GtkImage" id="webapp_icon">
-                        <property name="visible">True</property>
-                      </object>
-                    </child>
-                  </object>
-                  <packing>
-                    <property name="left-attach">0</property>
-                    <property name="top-attach">0</property>
-                    <property name="height">2</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkLabel">
-                    <property name="visible">True</property>
-                    <property name="halign">start</property>
-                    <property name="label" translatable="yes">Homepage:</property>
-                  </object>
-                  <packing>
-                    <property name="left-attach">1</property>
-                    <property name="top-attach">0</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkEntry" id="webapp_url">
-                    <property name="visible">True</property>
-                    <property name="hexpand">True</property>
-                    <signal name="changed" handler="on_webapp_entry_changed"/>
-                  </object>
-                  <packing>
-                    <property name="left-attach">2</property>
-                    <property name="top-attach">0</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkLabel">
-                    <property name="visible">True</property>
-                    <property name="halign">start</property>
-                    <property name="label" translatable="yes">Title:</property>
-                  </object>
-                  <packing>
-                    <property name="left-attach">1</property>
-                    <property name="top-attach">1</property>
-                  </packing>
-                </child>
+              <object class="GtkButton" id="webapp_icon_button">
+                <property name="visible">true</property>
+                <property name="valign">center</property>
+                <signal name="clicked" handler="on_webapp_icon_button_clicked"/>
+                <style>
+                  <class name="image-button"/>
+                </style>
                 <child>
-                  <object class="GtkEntry" id="webapp_title">
+                  <object class="GtkImage" id="webapp_icon">
                     <property name="visible">True</property>
-                    <property name="hexpand">True</property>
-                    <signal name="changed" handler="on_webapp_entry_changed"/>
                   </object>
-                  <packing>
-                    <property name="left-attach">2</property>
-                    <property name="top-attach">1</property>
-                  </packing>
-                </child>
+               </child>
+              </object>
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="HdyActionRow">
+            <property name="activatable">False</property>
+            <property name="title" translatable="yes">_Homepage</property>
+            <property name="use_underline">True</property>
+            <property name="visible">True</property>
+            <child>
+              <object class="GtkEntry" id="webapp_url">
+                <property name="visible">True</property>
+                <property name="valign">center</property>
+                <signal name="changed" handler="on_webapp_entry_changed"/>
+              </object>
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="HdyActionRow">
+            <property name="activatable">False</property>
+            <property name="title" translatable="yes">_Title</property>
+            <property name="use_underline">True</property>
+            <property name="visible">True</property>
+            <child>
+              <object class="GtkEntry" id="webapp_title">
+                <property name="visible">True</property>
+                <property name="valign">center</property>
+                <signal name="changed" handler="on_webapp_entry_changed"/>
               </object>
             </child>
           </object>
@@ -382,4 +357,11 @@
       </object>
     </child>
   </template>
+  <object class="GtkSizeGroup">
+    <property name="mode">horizontal</property>
+    <widgets>
+      <widget name="webapp_url"/>
+      <widget name="webapp_title"/>
+    </widgets>
+  </object>
 </interface>
